### PR TITLE
Expose torch.export.dynamic_dim() API

### PR DESCRIPTION
**Pull request by gmagogsfm**
With updated doc

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @voznesenskym @penguinwu @EikanWang @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @anijain2305

### CODE REVIEW

File: torch/export/__init__.py [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+import torch
+
+
+__all__ = [
+    "dynamic_dim",
+    "export",
+]
+
+def dynamic_dim(t: torch.Tensor, index: int):
```
Comment (gmagogsfm) [inline torch/export/__init__.py:11]: @avikchaudhuri please take a look at this part. Ignore the `export()` part for now, it is being reviewed in a different PR, my ghstack is acting up, so I can't push proper stacks now.

**Review by avikchaudhuri — APPROVED** [review]

**Pull request closed without merging**